### PR TITLE
[ADD] eCommerce: add to cart page

### PR DESCRIPTION
**Pull request by toaa-odoo**
Adding add to cart page to the eCommerce section. Forward to master.

New eCommerce structure: https://docs.google.com/spreadsheets/d/1O7Jg4T8kPE0HaLw-3YDjaj54MMk957-9akBFaSNKQA0/edit#gid=0

taskid: 3099603

### CODE REVIEW

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+To do so, go to :menuselection:`Configuration --> Settings --> Shop - Products` and tick
+:guilabel:`Prevent Sale of Zero Priced Product`. This creates a new :guilabel:`Button url` field
+where you can enter the **redirect URL** to be used. Then, set the price of the product to `0.00`
+either from the **product's template**, or from a
+:doc:`pricelist <../../../sales/sales/products_prices/prices/pricing>`.
+
+.. tip::
+   Hiding the :guilabel:`Add to Cart` button is often used by B2B eCommerces that need to restrict
+   purchases to customers with an account, but still want to display an online product catalog for
+   those without.
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:41]: Fair point :)

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+The **custom button** can be added on any pages of the website as an **inner content** building
+block, and is an *additional* button to the regular :guilabel:`Add to Cart` button.
+
+To add it, go on the **product page** of your choice, click :menuselection:`Edit --> Blocks` and
+place the building block. Once placed, you have the following options:
+
+- :guilabel:`Product`: select the product to link the button with. Selecting a product renders the
+  :guilabel:`Action` field available;
+- :guilabel:`Action`: choose if the button should :guilabel:`Add to Cart` or :guilabel:`Buy Now`
+  (instant checkout).
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:60]: No changes in your suggestion, so as discussed I'm assuming it is a mistake :)

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
+===========
+Add to cart
+===========
+
+There are multiple ways to add products to the **cart** and customize the actions of the
+**add-to-cart** button.
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst]: Changed it up a bit

**Review by chiaraprattico — APPROVED** [review]

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+   :alt: Contact us button on product page
+
+Custom 'Add to Cart' button
+===========================
+
+You can also create a custom :guilabel:`Add to Cart` button and link it to a specific product.
+The **custom button** can be added on any page of the website as an **inner content** building
+block, and is an *additional* button to the regular :guilabel:`Add to Cart` button.
+
+To add it, go on the **product page** of your choice, click :menuselection:`Edit --> Blocks` and
```
Comment (xpl-odoo) [inline content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:56]: You can add it to any page.

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  (instant checkout).
+
+.. image:: cart/cart-add.png
+   :align: center
+   :alt: Custom 'Add to Cart' button
+
+'Buy Now' button
+================
+
+You can enable the :guilabel:`Buy Now` button to instantly take the customer to **checkout** instead
```
Comment (xpl-odoo) [inline content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:71]:
```suggestion
You can enable the 'Buy Now' button to instantly take the customer to **checkout** instead
```

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+of adding the product to the cart. The :guilabel:`Buy Now` button is an *additional* button and
+does not replace the :guilabel:`Add to Cart` button. To enable it, go to
+:menuselection:`Configuration --> Settings --> Shop - Checkout Process` and tick
```
Comment (xpl-odoo) [inline content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:74]:
```suggestion
:menuselection:`Website --> Configuration --> Settings --> Shop - Checkout Process` and tick
```

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+.. image:: cart/cart-buy-now.png
+   :align: center
+   :alt: Buy Now button
+
+Re-order from portal
+====================
+
+Customers have the possibility to **re-order** items from **previous sales orders** on the customer
+portal. To do so, go to :menuselection:`Configuration --> Settings --> Shop - Checkout Process` and
```
Comment (xpl-odoo) [inline content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:85]:
```suggestion
portal. To do so, go to :menuselection:`Website --> Configuration --> Settings --> Shop - Checkout Process` and
```

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+enable :guilabel:`Re-order From Portal`. Customers can find the :guilabel:`Re-order` button on their
```
Comment (xpl-odoo) [inline content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:86]:
```suggestion
enable :guilabel:`Re-order From Portal`. Customers can find the :guilabel:`Order Again` button on their
```

**Review by xpl-odoo — CHANGES_REQUESTED** [review]

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 9]
```diff
@@ -0,0 +1,91 @@
+===========
+Add to cart
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:2]: I'm not sure I understand what you mean here. There's a new eCommerce structure, and it includes a checkout section :) : https://docs.google.com/spreadsheets/d/1O7Jg4T8kPE0HaLw-3YDjaj54MMk957-9akBFaSNKQA0/edit#gid=0

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+- (Add to cart and) redirect to the cart page;
+- (Add to cart and) open a dialog box to let the user choose.
+
+To configure the action button, go to :menuselection:`Configuration --> Settings --> Shop - Checkout
+Process`, and select one of the three options under the :guilabel:`Add to Cart` section.
+
+.. note::
+   The action of the :guilabel:`Add to Cart` button is specific per website.
+
+Hide the 'Add to Cart' button
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:26]: Good suggestion

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+=============================
+
+If you do not need or wish to use the :guilabel:`Add to Cart` button, you can replace it with a
+:guilabel:`Contact Us` button which integrates a custom **redirect URL** of your choice. When
+clicking on the :guilabel:`Contact Us` button, customers are taken to the specified **redirect
+URL**.
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:32]: True

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+To do so, go to :menuselection:`Configuration --> Settings --> Shop - Products` and tick
+:guilabel:`Prevent Sale of Zero Priced Product`. This creates a new :guilabel:`Button url` field
+where you can enter the **redirect URL** to be used. Then, set the price of the product to `0.00`
+either from the **product's template**, or from a
+:doc:`pricelist <../../../sales/sales/products_prices/prices/pricing>`.
+
+.. note::
+   Hiding the :guilabel:`Add to Cart` button is often used by B2B eCommerces that need to restrict
+   purchases only to customers with an account, but still want to display an online product catalog
+   for those without.
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:43]: It is intended for eCommerces that only sell B2B and not B2C, therefore actually choosing who can buy on their website by granting them portal access through a selection (from what I understood from the smartclass)

[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:43]: Same comment as above

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst (hunk 4 above)
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:56]: Reworded

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+
+.. image:: cart/cart-contactus.png
+   :align: center
+   :alt: Contact us button on product page
+
+Custom 'Add to Cart' button
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:49]: I changed it to "Customizable" which defines better what the button really is about, however if you have any suggestions, lmk as I don't know how else to name a button can be modified by the user.

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst (hunk 12 above)
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:43]: See explanation above, and I added a todo:: for when the doc containing the mandatory sign up" is published

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+- (Add to cart and) stay on the product page;
+- (Add to cart and) redirect to the cart page;
+- (Add to cart and) open a dialog box to let the user choose.
+
+To configure the action button, go to :menuselection:`Configuration --> Settings --> Shop - Checkout
+Process`, and select one of the three options under the :guilabel:`Add to Cart` section.
+
+.. note::
+   The action of the :guilabel:`Add to Cart` button is specific per website.
+
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:25]: Relevant comment, I removed the first note.

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 15]
```diff
@@ -0,0 +1,91 @@
+===========
+Add to cart
+===========
+
+The :guilabel:`Add to Cart` button can be customized in multiple ways to perform specific actions,
+disabled to prevent sales, or redirect to a specific URL. Additionally, it can be combined with two
+**extra buttons** such as the **"Buy Now" (instant checkout)** button and **"Re-order"** button to
+increase the number of 'buy' options for customers.
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:8]: Great suggestion

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+
+The :guilabel:`Add to Cart` button on the product page can be configured to also perform one the
+following actions, on top of **adding to cart**:
+
+- (Add to cart and) stay on the product page;
+- (Add to cart and) redirect to the cart page;
+- (Add to cart and) open a dialog box to let the user choose.
+
+To configure the action button, go to :menuselection:`Configuration --> Settings --> Shop - Checkout
+Process`, and select one of the three options under the :guilabel:`Add to Cart` section.
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:21]: I like the way you did it better

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+of your choice.
+
+.. note::
+   Hiding the :guilabel:`Add to Cart` button is often used by B2B eCommerces that need to restrict
+   purchases only to customers with an account, but still want to display an online product catalog
+   for those without.
+
+.. todo:: add :ref: to sign-up section doc when published to note
+
+To do so, go to :menuselection:`Website app --> Configuration --> Settings --> Shop - Products` and
```
Comment (xpl-odoo) [inline content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst:42]:
```suggestion
To do so, go to :menuselection:`Website --> Configuration --> Settings --> Shop - Products` and
```
It's not necessary to mention "app" - discussed that with Jon previously.

File: content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst (hunk 13 above)
[the remark this answers is not in the document]
Comment (xpl-odoo) [reply content/applications/websites/ecommerce/checkout_payment_shipping/cart.rst]: You can customize default/standard add to cart buttons as well, so I felt talking about "additional" buttons might have been clearer

**Comment by toaa-odoo** [comment]
Do *not* merge yet. To be included in mega PR for new eCommerce structure

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Comment by toaa-odoo** [comment]
@robodoo r+

**Pull request closed without merging**